mcp/mcpadm.sh

```diff
@@ -0,0 +1,803 @@
+        echo "SET-VERSION HELP:"
+        echo ""
+        echo "set-version changes to different version of service"
+        echo ""
+        echo "Usage: ${SCRIPT_NAME} set-version [options]"
+        echo ""
+        echo "Options:"
+        echo "  appengine        -  List versions of App Engine service and set version to choice"
+        echo "  cloudrun         -  List versions of Cloud Run service and set version to choice"
+        echo "  <version-name>   -  Set version as <version-name> if it exists"
+        echo ""
+        exit
+        ;;
+    esac
+  fi
+
+
+}
+
+installDependencies() {
+  echo ""
+  echo "Installing dependencies"
+
+    OS="$(uname)"
+    case ${OS} in
+        'Linux')
+            installLinuxDependencies
+            ;;
+        'Darwin')
+            installDarwinDependencies
+            ;;
+        *)
+            echo "Couldn't determine OS type."
+            return
+            ;;
+    esac
+}
+
+installLinuxDependencies() {
+  if  ! [ -x "$(command -v terraform)" ]
+  then
+    echo ""
+    echo "Getting terraform ..."
+    echo ""
+
+    cd "${BIN}"
+    TERRAFORM_URL_LIN=https://releases.hashicorp.com/terraform/0.12.0/terraform_0.12.0_linux_amd64.zip
+    TERRAFORM_FILE_LIN="${TERRAFORM_URL_LIN##*/}"
+    wget "${TERRAFORM_URL_LIN}"
+    unzip -o "${TERRAFORM_FILE_LIN}"
+    rm "${TERRAFORM_FILE_LIN}"
+    cd "${SCRIPT_DIR}"
+
+    echo ""
+    echo "Terraform for $OS installed."
+    echo ""
+  else
+    echo ""
+    echo "All dependencies installed"
+    echo ""
+  fi
+}
+
+installDarwinDependencies(){
+   # Install Terraform
+    if  ! [ -x "$(command -v terraform)" ]
+    then
+        echo ""
+        echo "Getting terraform ..."
+        echo ""
+
+        cd "${BIN}"
+        TERRAFORM_URL_DAR=https://releases.hashicorp.com/terraform/0.12.0/terraform_0.12.0_linux_amd64.zip
+        TERRAFORM_FILE_DAR="${TERRAFORM_URL_DAR##*/}"
+
+        echo "URL: ${TERRAFORM_URL_DAR}"
+        echo "File: ${TERRAFORM_FILE_DAR}"
+
+        curl "${TERRAFORM_URL_DAR}" --output "${TERRAFORM_FILE_DAR}"
+        unzip "${TERRAFORM_FILE_DAR}"
+        rm "${TERRAFORM_FILE_DAR}"
+        cd "${SCRIPT_DIR}"
+
+        echo ""
+        echo "Terraform for $OS installed."
+        echo ""
+    fi
+}
+
+runSetup(){
+  local choice backend autoapprove
+  if [ $# == 2 ] && [[ "$2" =~ (-)*(h)+(elp)? ]]
+  then
+    help "$1"
+  else
+    echo "terraform{" > main.tf
+    if [ $# -lt 2 ]
+    then
+      echo "Which backend will you use?"
+      select choice in "gcs" "s3" "http" "kubernetes" "local"; do
+        case $choice in
+          gcs )
+            gcsBackend
+            ;;
+          s3 )
+            s3Backend
+            ;;
+          http )
+            httpBackend
+            ;;
+          kubernetes )
+            kubernetesBackend
+            ;;
+          local)
+            localBackend
+            ;;
+          *);;
+        esac
+        break
+      done
+    else
+      backend=${2,,}
```

Comment:
unfortunately, these nice substitutions were introduced, I think, I Bash v4. Which even the latest Mac OS doesn't include. I suggest using `tr`